foursight_core/react/api/aws_stacks.py

```diff
@@ -0,0 +1,133 @@
+import boto3
+import json
+from .datetime_utils import convert_utc_datetime_to_useastern_datetime_string
+from .misc_utils import memoize
+from collections import OrderedDict
+from dcicutils.obfuscation_utils import obfuscate_dict
+from typing import Optional, Union
+
+
+_STACK_NAME_PREFIX = "c4-"
+
+@memoize
+def aws_get_stacks() -> list:
+    """
+    Returns the list of all known AWS CloudFormation stacks with various metadata.
+    """
+    stacks_info = []
+    c4 = boto3.resource("cloudformation")
+    for stack in sorted(c4.stacks.all(), key=lambda key: key.name):
+        if stack.name.startswith(_STACK_NAME_PREFIX):
+            stacks_info.append(_create_aws_stack_info(stack))
+    return stacks_info
+
+
+def _create_aws_stack_info(stack: object):
+    return {
+        "name": stack.name,
+        "id": stack.stack_id,
+        "description": stack.description,
+        "role_arn": stack.role_arn,
+        "status": stack.stack_status,
+        "updated": convert_utc_datetime_to_useastern_datetime_string(stack.last_updated_time),
+        "created": convert_utc_datetime_to_useastern_datetime_string(stack.creation_time)
+    }
+
+
+@memoize
+def aws_get_stack(stack_name: str) -> object:
+    """
+    Returns all detailed info for the given AWS CloudFormation stack name,
+    including various metadata, outputs, parameters, and resources.
+    Output and parameter values are obfuscated if their names represents a sensitive value.
+    """
+    result = {}
+    stack = _aws_get_stack_object(stack_name)
+    if stack:
+        result = {**_create_aws_stack_info(stack)}
```

Comment:
Is this different than
```
result = create_aws_stack_info(stack)
```
? Are you expecting some copying or type-casting action? If so at minimum a comment explaining that is in order. But this might just be redundant.

Reply:
Oh yeah, you're right; was remnant of something different I was doing previously.